src/openrct2/actions/TrackRemoveAction.hpp

```diff
@@ -0,0 +1,534 @@
+            {
+                entranceDirections = TrackSequenceProperties[trackType][0];
+            }
+
+            if (entranceDirections & TRACK_SEQUENCE_FLAG_ORIGIN && (tileElement->AsTrack()->GetSequenceIndex() == 0))
+            {
+                if (!track_remove_station_element(mapLoc.x, mapLoc.y, mapLoc.z / 8, _origin.direction, rideIndex, 0))
+                {
+                    return MakeResult(GA_ERROR::UNKNOWN, STR_RIDE_CONSTRUCTION_CANT_REMOVE_THIS, gGameCommandErrorText);
+                }
+            }
+
+            TileElement* surfaceElement = map_get_surface_element_at({ mapLoc.x, mapLoc.y });
+            if (surfaceElement == nullptr)
+            {
+                log_warning("Surface Element not found. x = %d, y = %d", mapLoc.x, mapLoc.y);
+                return MakeResult(GA_ERROR::UNKNOWN, STR_RIDE_CONSTRUCTION_CANT_REMOVE_THIS);
+            }
+
+            int8_t _support_height = tileElement->base_height - surfaceElement->base_height;
+            if (_support_height < 0)
+            {
+                _support_height = 10;
+            }
+
+            cost += (_support_height / 2) * RideTrackCosts[ride->type].support_price;
+        }
+
+        money32 price = RideTrackCosts[ride->type].track_price;
+        if (ride_type_has_flag(ride->type, RIDE_TYPE_FLAG_FLAT_RIDE))
+        {
+            price *= FlatRideTrackPricing[trackType];
+        }
+        else
+        {
+            price *= TrackPricing[trackType];
+        }
+        price >>= 16;
+        price = (price + cost) / 2;
+        if (ride->lifecycle_flags & RIDE_LIFECYCLE_EVER_BEEN_OPENED)
+            price *= -7;
+        else
+            price *= -10;
+
+        if (gGameCommandNestLevel == 1)
+        {
+            LocationXYZ16 coord;
+            coord.x = startLoc.x + 16;
+            coord.y = startLoc.y + 16;
+            coord.z = trackpieceZ;
+            network_set_player_last_action_coord(network_get_player_index(game_command_playerid), coord);
+        }
+
+        res->Cost = price;
+        return res;
+    }
+
+    GameActionResult::Ptr Execute() const override
+    {
+        auto res = MakeResult();
+        res->Position.x = _origin.x + 16;
+        res->Position.y = _origin.y + 16;
+        res->Position.z = _origin.z;
+        res->ExpenditureType = RCT_EXPENDITURE_TYPE_RIDE_CONSTRUCTION;
+
+        int16_t trackpieceZ = _origin.z;
+
+        // Stations require some massaging of the track type for comparing
+        auto comparableTrackType = _trackType;
+        switch (_trackType)
+        {
+            case TRACK_ELEM_BEGIN_STATION:
+            case TRACK_ELEM_MIDDLE_STATION:
+                comparableTrackType = TRACK_ELEM_END_STATION;
+                break;
+        }
+
+        bool found = false;
+        bool isGhost = GetFlags() & GAME_COMMAND_FLAG_GHOST;
+        TileElement* tileElement = map_get_first_element_at(_origin.x / 32, _origin.y / 32);
+        if (tileElement == nullptr)
```

Comment:
If you put a break into the do/while body when its null you can recycle the error for when: found == false